rules/Php81/NodeFactory/EnumFactory.php

```diff
@@ -117,12 +123,23 @@ private function createEnumCaseFromConst(ClassConst $classConst): EnumCase
     private function createEnumCaseFromDocComment(
         PhpDocTagNode $phpDocTagNode,
         Class_ $class,
-        array $mapping = []
+        array $mapping = [],
+        bool $enumNameInSnakeCase = false,
     ): EnumCase {
         /** @var MethodTagValueNode $nodeValue */
         $nodeValue = $phpDocTagNode->value;
         $enumValue = $mapping[$nodeValue->methodName] ?? $nodeValue->methodName;
-        $enumName = strtoupper($nodeValue->methodName);
+        if ($enumNameInSnakeCase) {
+            $enumName = strtoupper(
+                Strings::replace(
+                    $nodeValue->methodName,
+                    '/(?<=[A-Z])(?=[A-Z][a-z])|(?<=[^A-Z])(?=[A-Z])|(?<=[A-Za-z])(?=[^A-Za-z])/',
```

Comment:
could you move it to constant and add `@see https://regex101.com/` uri to show the match, like other usage, eg:

https://github.com/rectorphp/rector-src/blob/a8cc19a3c0387f082fd432e04afce363a90ba5b0/rules/CodeQuality/Rector/Concat/JoinStringConcatRector.php#L26-L31